### PR TITLE
fix: update types, use correct arguments for listen_callback actions

### PR DESCRIPTION
**Pull request by HipsterBrown**
While working on adding this module to my smart assistant project, I ran into some issues related to the class methods called from the `listen_callback` and invalid types. I've resolved those issues in this PR and tested on my smart machine.

### CODE REVIEW

File: src/speech/speechio.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 from viam.proto.common import ResourceName
 from viam.resource.base import ResourceBase
 from viam.resource.types import Model
-from viam import logging
+from viam.logging import getLogger
```
Comment (HipsterBrown) [inline src/speech/speechio.py:15]: Updated to be consistent with other Viam code.

File: src/speech/speechio.py [hunk 2]
```diff
@@ -23,25 +23,26 @@
...
 from speech_service_api import SpeechService
 
-LOGGER = logging.getLogger(__name__)
-CACHEDIR = "/tmp/cache"
-
-mixer.init(buffer=1024)
-rec_state = {}
-
-class SpeechProvider(Enum):
+class SpeechProvider(str, Enum):
```
Comment (HipsterBrown) [inline src/speech/speechio.py:26]: Needs to inherit from `str` as well to be JSON serializable ([reference](https://hultner.se/quickbits/2018-03-12-python-json-serializable-enum.html)).

File: src/speech/speechio.py [hunk 3]
```diff
@@ -74,7 +75,7 @@ class SpeechIOService(SpeechService, Reconfigurable):
     @classmethod
     def new(cls, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]) -> Self:
         speechio = cls(config.name)
-        speechio = speechio.reconfigure(config, dependencies)
+        speechio.reconfigure(config, dependencies)
```
Comment (HipsterBrown) [inline src/speech/speechio.py:78]: `reconfigure` should return `None` according to the `Reconfigurable` class. Updated to match expected return type.

File: src/speech/speechio.py [hunk 4]
```diff
@@ -87,7 +88,7 @@ async def say(self, text: str, blocking: bool, cache_only: bool = False) -> str:
         if not os.path.isdir(CACHEDIR):
             os.mkdir(CACHEDIR)
 
-        file = CACHEDIR + '/' + self.speech_provider + self.speech_voice + self.completion_persona + hashlib.md5(text.encode()).hexdigest() + ".mp3"
+        file = os.path.join(CACHEDIR, self.speech_provider.value + self.speech_voice + self.completion_persona + hashlib.md5(text.encode()).hexdigest() + ".mp3")
```
Comment (HipsterBrown) [inline src/speech/speechio.py:91]: Updated to use safe path building.

File: src/speech/speechio.py [hunk 5]
```diff
@@ -179,18 +182,18 @@ def listen_callback(self, recognizer, audio):
...
-            transcript = recognizer.recognize_google(audio,show_all=True)
+            transcript = recognizer.recognize_google(audio, show_all=True)
             if type(transcript) is dict and transcript.get("alternative"):
                 heard = transcript["alternative"][0]["transcript"]
                 LOGGER.debug("speechio heard " + heard)
                 if (self.listen and re.search(".*" + self.listen_trigger_say, heard))or (self.trigger_active and self.active_trigger_type == 'say'):
                     self.trigger_active = False
                     to_say = re.sub(".*" + self.listen_trigger_say + "\s+",  '', heard)
-                    asyncio.run(self.say(to_say))
+                    asyncio.run(self.say(to_say, blocking=False))
```
Comment (HipsterBrown) [inline src/speech/speechio.py:192]: `blocking` is a required argument for `say` and `completion`.

File: src/speech/speechio.py [hunk 6]
```diff
@@ -200,17 +203,18 @@ def listen_callback(self, recognizer, audio):
...
+                    if rec_state.listen_closer != None:
+                        rec_state.listen_closer()
         except sr.UnknownValueError:
             LOGGER.warn("Google Speech Recognition could not understand audio")
         except sr.RequestError as e:
             LOGGER.warn("Could not request results from Google Speech Recognition service; {0}".format(e))
 
     def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]):
-        self.speech_provider = config.attributes.fields["speech_provider"].string_value or 'google'
+        self.speech_provider = SpeechProvider[config.attributes.fields["speech_provider"].string_value or "google"]
```
Comment (HipsterBrown) [inline src/speech/speechio.py:214]: Actually using the class enum.

**Review by mcvella — APPROVED** [review]
lgtm ty

**Pull request merged**